bcda/web/middleware/validation.go

```diff
@@ -0,0 +1,173 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/CMSgov/bcda-app/bcda/responseutils"
+	fhircodes "github.com/google/fhir/go/proto/google/fhir/proto/stu3/codes_go_proto"
+)
+
+var supportedOutputFormats = map[string]struct{}{
+	"ndjson":                  {},
+	"application/fhir+ndjson": {},
+	"application/ndjson":      {}}
+
+type RequestParameters struct {
+	Since         time.Time
+	ResourceTypes []string
+	Version       string // e.g. v1, v2
+}
+
+// requestkey is an unexported context key to avoid collisions
+type requestkey int
+
+var rk requestkey
```

Comment:
Make this a const? Just in case

Reply:
great call out! I misread Go's guidance here and was using a var. Ref: https://blog.golang.org/context